commit fb11a30427c2a63d6420ec4eda5d1d816cba5af2
Author: Marc Zyngier <marc.zyngier@arm.com>
Date:   Fri Feb 17 14:32:18 2017 +0000

    arm64: KVM: VHE: Clear HCR_TGE when invalidating guest TLBs
    
    commit 68925176296a8b995e503349200e256674bfe5ac upstream.
    
    When invalidating guest TLBs, special care must be taken to
    actually shoot the guest TLBs and not the host ones if we're
    running on a VHE system.  This is controlled by the HCR_EL2.TGE
    bit, which we forget to clear before invalidating TLBs.
    
    Address the issue by introducing two wrappers (__tlb_switch_to_guest
    and __tlb_switch_to_host) that take care of both the VTTBR_EL2
    and HCR_EL2.TGE switching.
    
    Reported-by: Tomasz Nowicki <tnowicki@caviumnetworks.com>
    Tested-by: Tomasz Nowicki <tnowicki@caviumnetworks.com>
    Reviewed-by: Christoffer Dall <cdall@linaro.org>
    Signed-off-by: Marc Zyngier <marc.zyngier@arm.com>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>

Index: linuxtest/dummy/rpi_1452_fb11a30427c2a63d6420ec4eda5d1d816cba5af2.txt
===================================================================
--- /dev/null
+++ linuxtest/dummy/rpi_1452_fb11a30427c2a63d6420ec4eda5d1d816cba5af2.txt
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
